v4/newrelic/segments.go

```diff
@@ -230,16 +231,38 @@ func (s *DatastoreSegment) End() {
 	s.StartTime.end()
 }
 
+var (
+	hostsToReplace = map[string]struct{}{
```

Comment:
It seemed a little weird to me at first to use a map with empty values for this, but I can see how that would be the fastest and best way. My impulse was to use a Set, but that's a Java thing, not a Go thing. Still getting rid of my old habits.

Reply:
Yeah, Go doesn't have Sets, but it does have maps with empty values. And for the record, I copied this from the v3 code anyway.